README.md

```diff
@@ -1,37 +1,104 @@
 
-Details:
+Following steps go over the starting the scheduler in PowerOf2 mode and start 4 workers (2 of them with GPU). 
 
 1) Install miniconda from https://docs.anaconda.com/free/miniconda/
-2) conda create --name cs244b --file requirements.txt python=3.9
-3) conda activate cs244b
-4) Run compile_proto script based on your target machine from root repo folder (CS244B)
-5) Start the scheduler server from one terminal - 
-    python3 scheduler_server.py --PortNumber 50051 --SchedulerMode Random
-6) Start multiple worker servers in another terminal - python3 worker_server.py --PortNumber <PortNumber> --MaxThreadCount <MaxThreadCount> --HardwareGeneration <HardwareGeneration>
-    Example Command:
-    python3 worker_server.py --PortNumber 50055 --MaxThreadCount 2 --HardwareGeneration Gen2 --gpuEnabled
-
-    First parameter is the port number
-    Second parameter is the number of threads the worker can process
-    Third parameter is the hardware generation parameter which determines if scheduler would send more requests to this worker in LoadAware scheduling mode
-    Fourth parameter is an optional parameter. If --gpuEnabled parameter is passed, worker is treated as GPU Enabled.
-    Fifth parameter is an optional parameter. If --addDelay parameter is passed, delays are added randomly for the worker operations.
-
-7) Run driver for another terminal - python3 driver.py (There is an example code written under main for testing)
-8) Submit tasks with the driver; they should be seen getting executed on the worker.
+
+2) Create a conda environment for the experiments
+```
+conda create --name cs244b --file requirements.txt python=3.9
+```
+
+3) Activate the conda package and environment manager:
+```
+conda activate cs244b
+```
+
+4) Run compile_proto script from root repo folder [CS244B](https://github.com/krish8484/cs244b) which generates the necessary gRPC proto files. The script depends on your target machine. 
+
+For UNIX based machines, run:
+```
+./compile_proto.sh
+```
+
+For Windows OS machine, run:
+```
+compile_proto.cmd
+```
+
+5) Start the scheduler server from a terminal -
+```
+cd Scheduler;
+python3 scheduler_server.py --PortNumber 50051 --SchedulerMode PowerOf2
+```
+Reference Command:
+`python3 scheduler_server.py --PortNumber <PortNumber> --SchedulerMode <SchedulingMode> --AssingedWorkersPerTask <AssignedWorkersPerTask>`
+
+- First parameter is the port number. Type `int`.
+- Second parameter is the scheduling algorithm that would be used for request routing in the scheduler. Type `string`. It supports 4 modes `Random`, `RoundRobin`, `LoadAware` and `PowerOf2`
+- Third parameter is the number of workers a single task would assigned.  Type `int`. This Task Replication Strategy helps up retrieve a task computation from another worker when one of them is down.
```

Comment:
Please include 3rd param is optional, and it defaults to 1 which is no FT

Reply:
can be taken up in a subsequent PR if needs to be done